From: Sam Hocevar <sam@hocevar.net>
Subject: [PATCH] Link vlc-cache-gen with the C++ standard library to avoid
 issues when we dlclose() a C++ plugin that was built with a
 faulty library that may have registered __cxa_atexit
 callbacks (usually through the libstdc++ that library was
 built with).
 .
 Upgrading libstdc++ is not always a solution since some code may have been
 inlined and we can't get rid of it.
Origin: upstream, http://git.videolan.org/?p=vlc.git;a=commitdiff;h=07d38617832be934896fa7a14fe52f1a65c901ab
Bug-Debian: http://bugs.debian.org/660934
Bug-Debian: http://bugs.debian.org/660935

--- a/bin/Makefile.am
+++ b/bin/Makefile.am
@@ -59,7 +59,7 @@
 vlc_win32_rc.$(OBJEXT): vlc_win32_rc.rc
 	$(WINDRES) --include-dir $(top_srcdir)/share/icons -i $< -o $@
 
-vlc_cache_gen_SOURCES = cachegen.c
+vlc_cache_gen_SOURCES = cachegen.c dummy.cpp
 vlc_cache_gen_LDADD = \
 	$(GNUGETOPT_LIBS) \
 	../compat/libcompat.la \
--- /dev/null
+++ b/bin/dummy.cpp
@@ -0,0 +1 @@
+/* This file is only here to force cachegen to be linked against libstdc++ */
